### ISSUE TITLE
laz-perf (included in qgis 3.22) fails to build on NetBSD

### ISSUE THREAD

**Issue by gdt** [hobu/laz-perf#115]
On NetBSD 9, I got complains of redefitniions of many macros like le32toh and finally an undefined variable.

Reading [portable_endian.h](https://github.com/hobu/laz-perf/tree/master/cpp/lazperf/portable_endian.hpp) I see that NetBSD is treated like Dragonfly and there is renaming, while on FreeBSD and OpenBSD sys/endian.h is simply included.   Only NetBSD 8 and later matter, but NetBSD got sys/endian.h in 1999, and it doesn't seem to have definitions for the target symbols in the Dragonfly case.

I'm guessing this was just an incorrect guess that NetBSD was like DragonFly, when really it seems all are similar except for DragonFly.  I'm happy to submit a PR; the following diff (which github web UI refuses to attach) caused the qgis build to make vastly more progress (still running):

```
--- portable_endian.hpp.orig    2022-01-14 07:06:34.000000000 -0500
+++ portable_endian.hpp 2022-02-02 20:11:56.651221385 -0500
@@ -43,11 +43,11 @@
 #   define __PDP_ENDIAN    PDP_ENDIAN
 **/
 
-#elif defined(__OpenBSD__)|| defined(__FreeBSD__) 
+#elif defined(__OpenBSD__)|| defined(__FreeBSD__) || defined(__NetBSD__)
 
 #   include <sys/endian.h>
 
-#elif defined(__NetBSD__) || defined(__DragonFly__)
+#elif defined(__DragonFly__)
 
 #   define be16toh betoh16
 #   define le16toh letoh16
garlic gdt 554 ~ > mv D laz-perf-endian-fix.diff
garlic gdt 555 ~ > mv laz-perf-endian-fix.diff  laz-perf-endian-fix.txt
garlic gdt 556 ~ > p laz-perf-endian-fix.txt 
--- portable_endian.hpp.orig    2022-01-14 07:06:34.000000000 -0500
+++ portable_endian.hpp 2022-02-02 20:11:56.651221385 -0500
@@ -43,11 +43,11 @@
 #   define __PDP_ENDIAN    PDP_ENDIAN
 **/
 
-#elif defined(__OpenBSD__)|| defined(__FreeBSD__) 
+#elif defined(__OpenBSD__)|| defined(__FreeBSD__) || defined(__NetBSD__)
 
 #   include <sys/endian.h>
 
-#elif defined(__NetBSD__) || defined(__DragonFly__)
+#elif defined(__DragonFly__)
 
 #   define be16toh betoh16
 #   define le16toh letoh16

```

I'm happy to submit a PR if you are inclined to take it.

**Comment by hobu** [comment]
Yes we would be happy to merge something. There will be a new release soon to address the Apache Public License relicencing. 

This code will also be vendored in PDAL as well. I don't know if PDAL is your ultimate user of this or not.

**Comment by gdt** [comment]
OK, pull request soon.   I am here because the vendored code in qgis failed to build.

I am surprised about how DragonFly is handled, because sys/endian.h in
the BSD world seems like it predates the FreeBSD/DragonFly split, but
I don't use DragonFly.  I don't want to entangle the NetBSD fix with that issue, though.

**Comment by hobu** [comment]
This code is also vendored in PDAL https://github.com/PDAL/PDAL/blob/e416c45c1d0d08d0610cbc707237e97d0f7741f9/pdal/util/portable_endian.hpp so fixes or changes here will need to be pushed up there too.

**Comment by gdt** [comment]
I hope there is some semi-automated update process, to mitigate the maintenance problems inherent in vendoring.  Is that meant as note-to-self, or are you suggesting I open a PDAL and qgis bug report that (once #116 merged) the vendored code is not up to date?

**Comment by hobu** [comment]
Mostly as a note to self.

Although, as I mentioned, we are going to be vendoring laz-perf into PDAL. This was described in a mail to the PDAL list https://lists.osgeo.org/pipermail/pdal/2021-October/002483.html With LASzip and laz-perf now being relicensed as APL #114, we can vendor it and make LAZ support a much easier and simple thing for PDAL's users.
